main.go

```diff
@@ -59,51 +69,60 @@ func downloadFileToBuffer(url string, concurrency int) (*bytes.Buffer, error) {
 		go func(start, end int64) {
 			defer wg.Done()
 
-			retries := 5
-			for retries > 0 {
+			success := false
+			for retryNum := 0; retryNum < retries; retryNum++ {
 				transport := http.DefaultTransport.(*http.Transport).Clone()
 				transport.DisableKeepAlives = true
 				client := &http.Client{
 					Transport: transport,
 				}
 
 				req, err := http.NewRequest("GET", url, nil)
+				sleepJitter := time.Duration(rand.Intn(retrySleepJitter))
+				sleepTime := time.Millisecond * (sleepJitter + retryDelayBaseline)
+				if retryNum > 0 {
+					// Exponential backoff
+					// 2^retryNum * retryBackoffIncr (in milliseconds)
+					backoffFactor := math.Pow(retryBackoffFactor, float64(retryNum))
+					backoffDuration := time.Duration(math.Min(backoffFactor*retryBackoffIncr, retryMaxBackoffTime))
+					sleepTime += (time.Millisecond * backoffDuration)
+				}
 				if err != nil {
+					// This needs to be a time.Duration to make everything happy
 					fmt.Printf("Error creating request: %v\n", err)
-					retries--
-					time.Sleep(time.Millisecond * 100) // wait 100 milliseconds before retrying
+					time.Sleep(sleepTime) // wait sleepTime milliseconds before retrying
 					continue
 				}
 				req.Header.Set("Range", fmt.Sprintf("bytes=%d-%d", start, end))
+				if retryNum > 0 {
+					req.Header.Set("Pget-Retry-Count", fmt.Sprintf("%d", retryNum))
```

Comment:
```suggestion
					req.Header.Set("Retry-Count", fmt.Sprintf("%d", retryNum))
```

just call it retry-count